client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx

```diff
@@ -48,27 +53,29 @@ const ProcessingStep: Step = function ( props ) {
 	const captureFlowException = useCaptureFlowException( props.flow, 'ProcessingStep' );
 
 	useEffect( () => {
-		( async () => {
-			if ( typeof action === 'function' ) {
-				try {
-					const destination = await action();
-					// Don't call submit() directly; instead, turn on a flag that signals we should call submit() next.
-					// This allows us to call the newest submit() created. Otherwise, we would be calling a submit()
-					// that is frozen from before we called action().
-					// We can now get the most up to date values from hooks inside the flow creating submit(),
-					// including the values that were updated during the action() running.
-					setDestinationState( destination );
-					setHasActionSuccessfullyRun( true );
-				} catch ( e ) {
-					// eslint-disable-next-line no-console
-					console.error( 'ProcessingStep failed:', e );
-					captureFlowException( e );
-					submit?.( {}, ProcessingResult.FAILURE );
+		if ( action ) {
```

Comment:
Why do we need to check action here?

Reply:
I'm not sure if it still needed. It was a patch added before
`submit?.( {}, ProcessingResult.NO_ACTION );` was added, maybe. And it would have covered users hitting the back button from checkout. If we test it and it is not needed, then we can remove it 👍

Reply:
On a second thought, I think it was added because otherwise it would be difficult for users to go back in the flow after reaching checkout.